### DISCUSSION TITLE
Documentation: DB_SEARCH_PATH & multiple PostgreSQL schema support

### DISCUSSION THREAD

**Discussion by peteruithoven** [directus/directus#5395] [Archive]
We thought Directus didn't support multiple PostgreSQL schema's until we started digging through issues and discussions like:
- https://github.com/directus/directus/discussions/3164
- https://github.com/directus/directus/issues/4322
- https://github.com/directus/directus/issues/3228

So Directus does seem to support multiple PostgreSQL schema's through the `DB_SEARCH_PATH` environment variable?

This is kind of explained by this comment:
> All DB_* environment variables are passed to the connection configuration of a Knex instance (opens new window). Based on your project's needs, you can extend the DB_* environment variables with any config you need to pass to the database instance.

From: https://docs.directus.io/reference/environment-variables/#database

But seeing the questions this might be worth mentioning and or adding an example for?

**Comment by rijkvanzanten** [comment]
You can rely on search-path on the connection level to search through multiple schemas. That being said, Directus itself doesn't actually do anything with schema's (doesn't prefix tables in sql queries etc) at this point, so your mileage may very. I wouldn't rely on search path for the sake of using _multiple_ schemas in the same Directus instance at this time, but it should be fine to scope your instance on a single schema.

The reason it's not documented explicitly under env vars right now is because Directus doesn't support (multiple) schemas across database vendors 👍🏻 

Groetjes!

**Reply by peteruithoven** [reply]
Out of curiosity, would this fall under the 80% of the 80/20 Rule?

> To keep the Directus codebase as clean and simple as possible, we will only consider approving features that we feel at least 80% of our user-base will find valuable. If your feature request falls within the 20% range, it is considered an edge-case and should be implemented as an extension.

**Reply by rijkvanzanten** [reply]
Yeah totally. I've been meaning to add "native" support for selecting the schema to operate on by relying on Knex's `withSchema` in queries where it's schema appropriate, but it's a little tricky to just tack on, as there's no way to set that globally for every query as of right now.

What you're asking for though—the ability to have the Directus tables in a different schema than the user tables—falls under a bigger question around multi-datastore support which we're actively investigating for future versions, but as you can imagine, that's a way bigger endeavor 👍🏻

**Comment by ilijaNL** [comment]
Another use-case for this is feature is that you can restrict permissions on schema level. Sometimes you want to use directus only as a admin panel without having permissions to modify any data structure (especially during database mirroring), since i don't see any other approach. I think hasura handles this use case pretty well (graphql only). And adding support for multiple database also gives an ability to keep one database pure for configuration of directus.
